controller/environment_space_scope_blackbox_test.go

```diff
@@ -0,0 +1,199 @@
+package controller_test
+
+import (
+	"context"
+	"crypto/rsa"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/fabric8-services/fabric8-common/service"
+	testauth "github.com/fabric8-services/fabric8-common/test/auth"
+	testsuite "github.com/fabric8-services/fabric8-common/test/suite"
+	"github.com/fabric8-services/fabric8-common/token"
+	"github.com/fabric8-services/fabric8-env/app"
+	"github.com/fabric8-services/fabric8-env/app/test"
+	"github.com/fabric8-services/fabric8-env/configuration"
+	"github.com/fabric8-services/fabric8-env/controller"
+	"github.com/fabric8-services/fabric8-env/gormapp"
+	"github.com/goadesign/goa"
+	uuid "github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+/*
+Setup:
+
+One space 	=> space1
+Three users	=> user1, user2, user3
+
+** user_space_scope matrix **
+================================
+user 	manage 	contribute 	view
+================================
+user1	yes		yes			yes
+user2	no		yes			yes
+user3	no		no			yes
+
+** user operation matrix **
+================================
+user 	create 	list		show
+================================
+user1	yes		yes			yes
+user2	no		yes			yes
+user3	no		no			no
+*/
+
+var testUser1 = &testauth.Identity{ID: uuid.NewV4(), Email: "user1@test.com", Username: "user1"} // user1
+var testUser2 = &testauth.Identity{ID: uuid.NewV4(), Email: "user2@test.com", Username: "user2"} // user2
+var testUser3 = &testauth.Identity{ID: uuid.NewV4(), Email: "user3@test.com", Username: "user3"} // user3
+
+type EnvironmentSpaceScopeSuite struct {
+	testsuite.DBTestSuite
+	db *gormapp.GormDB
+
+	svc        *goa.Service
+	ctrl       *controller.EnvironmentController
+	authServer *httptest.Server
+
+	ctx1      context.Context
+	ctx2      context.Context
+	ctx3      context.Context
+	spaceID   uuid.UUID // space1
+	publicKey *rsa.PublicKey
+}
+
+func TestEnvironmentSpaceScopeSuite(t *testing.T) {
+	config, err := configuration.New("")
+	require.NoError(t, err)
+	suite.Run(t, &EnvironmentSpaceScopeSuite{DBTestSuite: testsuite.NewDBTestSuite(config)})
+}
+
+func (s *EnvironmentSpaceScopeSuite) SetupSuite() {
+	s.DBTestSuite.SetupSuite()
+
+	s.db = gormapp.NewGormDB(s.DB)
+	s.authServer = s.startAuthServer()
+	authURL := "http://" + s.authServer.Listener.Addr().String() + "/"
```

Comment:
you could use `fmt.Sprintf("https://%v/", s.authServer.Listener.Addr())` instead

Reply:
thanks, I will take care next time.